sdk/core/azure-core/src/main/java/com/azure/core/implementation/util/ListByteBufferContent.java

```diff
@@ -0,0 +1,108 @@
+package com.azure.core.implementation.util;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.serializer.ObjectSerializer;
+import com.azure.core.util.serializer.TypeReference;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.io.InputStream;
+import java.nio.Buffer;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicReferenceFieldUpdater;
+
+public class ListByteBufferContent extends BinaryDataContent {
+    private static final ClientLogger LOGGER = new ClientLogger(ListByteBufferContent.class);
+
+    private final List<ByteBuffer> content;
+
+    private volatile byte[] bytes;
+    private static final AtomicReferenceFieldUpdater<ListByteBufferContent, byte[]> BYTES_UPDATER
+        = AtomicReferenceFieldUpdater.newUpdater(ListByteBufferContent.class, byte[].class, "bytes");
+
+    private Long cachedLength;
+
+    /**
+     * Creates a new instance of {@link BinaryDataContent}.
+     *
+     * @param content The {@link ByteBuffer} content.
+     * @throws NullPointerException If {@code content} is null.
+     */
+    public ListByteBufferContent(List<ByteBuffer> content) {
+        this.content = Objects.requireNonNull(content, "'content' cannot be null.");
+    }
+    @Override
+    public Long getLength() {
+        if (cachedLength == null) {
+            cachedLength = content.stream().mapToLong(Buffer::remaining).sum();
+        }
+        return cachedLength;
+    }
+
+    @Override
+    public String toString() {
+        return new String(toBytes(), StandardCharsets.UTF_8);
+    }
+
+    @Override
+    public byte[] toBytes() {
+        return BYTES_UPDATER.updateAndGet(this, bytes -> bytes == null ? getBytes() : bytes);
+    }
+
+    @Override
+    public <T> T toObject(TypeReference<T> typeReference, ObjectSerializer serializer) {
+        return serializer.deserializeFromBytes(toBytes(), typeReference);
+    }
+
+    @Override
+    public InputStream toStream() {
+        return new IterableOfByteBuffersInputStream(content);
+    }
+
+    @Override
+    public ByteBuffer toByteBuffer() {
+        return ByteBuffer.wrap(toBytes()).asReadOnlyBuffer();
+    }
+
+    @Override
+    public Flux<ByteBuffer> toFluxByteBuffer() {
+        return Flux.fromIterable(content).map(ByteBuffer::asReadOnlyBuffer);
+    }
+
+    @Override
+    public boolean isReplayable() {
+        return true;
+    }
+
+    @Override
+    public BinaryDataContent toReplayableContent() {
+        return this;
+    }
+
+    @Override
+    public Mono<BinaryDataContent> toReplayableContentAsync() {
+        return Mono.just(this);
+    }
+
+    private byte[] getBytes() {
+        long length = getLength();
```

Comment:
I guess we can skip this call to `getLength()` and keep a running sum of the length in the for loop below and if the sum exceeds the max array size, throw an exception. We don't have to go through the list of all byte buffers twice.

Reply:
Since we need the length to allocate the byte array and not resize the array and copy, let's leave this as is.